lib/env.ts

```diff
@@ -0,0 +1,65 @@
+import { Option, fromNullable } from "fp-ts/lib/Option";
+import * as t from "io-ts";
+
+export const EnvKey = t.union([
+  t.literal("SPID_SESSION_TOKEN"),
+  t.literal("IO_BACKEND_HOST"),
+  t.literal("IO_BACKEND_BASEPATH")
+]);
+
+export type EnvKey = t.TypeOf<typeof EnvKey>;
+
+export const EnvMap = t.record(EnvKey, t.string);
+export type EnvMap = Partial<t.TypeOf<typeof EnvMap>>;
+
+const pick = (obj: any) =>
```

Comment:
naming: it looks like a generic method but at the same time it works only with EnvMap / EnvKey. a more specific method and a comment would clarify what this method is supposed to do

Reply:
you're right, it was originally named `pickEnvKeys`, but I crewed that after refactor 😅

Reply:
Anyway, this seems a more general problem, does `io-ts` have any helper to deal with it?

Reply:
to do what exactly ?

Reply:
Basically what I did with my custom `pickEnvKeys` function.

Reply:
it's not very clear from the typings (since it takes an "any" and the return type is not specified). but I guess it's supposed to extract some values, identified by a list of literals, from process.env.

What about `EnvMap.decode(process.env).map(config => ...)` <-- you have all the needed values here

Reply:
`EnvMap.decode(process.env)` would fail, as `process.env` has may more keys. Anyway, we can live without that for now

Reply:
it won't

Reply:
I'd suggest to drop the whole module and use something like

Reply:
this fails:

```
const toEnvMap = (obj: any) =>
  EnvMap.decode(obj).getOrElseL(() => {
    throw new Error("decode failed");
  });
const envMap = toEnvMap(process.env); // err: decoded failed
```

Reply:
that's because it's a record, interface have a loose validation

Reply:
yes, but still I'm getting **all** environment variables if I merge the object. That is: with interface, what I get at runtime is not what I get in the type system

Reply:
why / when do you need to merge the object ? (which object ?)

Reply:
I'm getting defaults from `process.env` at startup, then I integrate what's missing with values from prompt. 
So I'm doing `const env: EnvMap = {...process.env}`, but that brings a lot of other variables. So the need for `pick` function

Reply:
Now I get it :-) In this case you can use t.exact:

```ts

import * as t from "io-ts";

const ConfigT = t.exact(t.interface({ HOME: t.string, USERNAME: t.string }));

const config = ConfigT.decode(process.env).getOrElseL(() => {
  throw Error("...");
});

console.log(config);

```